plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx

```diff
@@ -0,0 +1,191 @@
+		return {
+			text: __( 'Expired', 'woocommerce' ),
+			warning: true,
+			explanation: __(
+				'To receive updates and support for this extension, you need to purchase a new subscription or consolidate your extensions to one connected account by sharing or transferring this extension to this connected account.',
+				'woocommerce'
+			),
+		};
+	}
+
+	return {
+		text: __( 'Inactive', 'woocommerce' ),
+		warning: true,
+		explanation: __(
+			'To receive updates and support for this extension, you need to purchase a new subscription or consolidate your extensions to one connected account by sharing or transferring this extension to this connected account.',
+			'woocommerce'
+		),
+	};
+}
+
+function getVersion( subscription: Subscription ): string {
+	if ( subscription.local.version === subscription.version ) {
+		return subscription.local.version;
+	}
+
+	if ( subscription.local.version && subscription.version ) {
+		return subscription.local.version + ' > ' + subscription.version;
+	}
+
+	if ( subscription.version ) {
+		return subscription.version;
+	}
+
+	if ( subscription.local.version ) {
+		return subscription.local.version;
+	}
+
+	return '';
+}
+
+export function productName( subscription: Subscription ): TableRow {
+	// This is the fallback icon element with products without an icon.
+	let iconElement = <Icon icon={ plugins } size={ 40 } />;
+
+	// If the product has an icon, use that instead.
+	if ( subscription.product_icon ) {
+		iconElement = (
+			<img
+				src={ subscription.product_icon }
+				alt={ sprintf(
+					/* translators: %s is the product name. */
+					__( '%s icon', 'woocommerce' ),
+					subscription.product_name
+				) }
+				className="woocommerce-marketplace__my-subscriptions__product-icon"
+			/>
+		);
+	}
+
+	const displayElement = (
+		<div className="woocommerce-marketplace__my-subscriptions__product">
+			{ iconElement }
+			<span className="woocommerce-marketplace__my-subscriptions__product-name">
+				{ subscription.product_name }
+			</span>
+		</div>
+	);
+
+	return {
+		display: displayElement,
+		value: subscription.product_name,
+	};
+}
+
+export function status( subscription: Subscription ): TableRow {
+	const subscriptionStatus = getStatus( subscription );
+
+	let statusElement = <>{ subscriptionStatus.text }</>;
+
+	if ( subscriptionStatus.warning ) {
+		statusElement = (
+			<StatusPopover
+				text={ subscriptionStatus.text }
+				explanation={ subscriptionStatus.explanation ?? '' }
+			/>
+		);
+	}
+
+	const displayElement = (
+		<span className="woocommerce-marketplace__my-subscriptions__status">
+			{ statusElement }
+		</span>
+	);
+
+	return {
+		display: displayElement,
+		value: subscriptionStatus.text,
+	};
+}
+
+export function expiry( subscription: Subscription ): TableRow {
+	const expiryDate = subscription.expires;
+
+	let expiryDateElement = __( 'Never expires', 'woocommerce' );
+
+	if ( expiryDate ) {
+		expiryDateElement = gmdateI18n(
+			'j M, Y',
```

Comment:
Do we need the comma? Looks weird to me.

```suggestion
			'j M Y',
```

Reply:
The designs have a comma. Looking at the orders page, I think it's a WooCommerce format that we follow. So I'll keep this as is.